consensus/istanbul/backend/backend.go

```diff
@@ -366,22 +372,22 @@ func (sb *Backend) SendDelegateSignMsgToProxiedValidator(msg []byte) error {
 
 // Authorize implements istanbul.Backend.Authorize
 func (sb *Backend) Authorize(ecdsaAddress, blsAddress common.Address, publicKey *ecdsa.PublicKey, decryptFn istanbul.DecryptFn, signFn istanbul.SignerFn, signBLSFn istanbul.BLSSignerFn, signHashFn istanbul.HashSignerFn) {
```

Comment:
It would be nice if  we could do this at contstruction time, then we wouldn't need the synchronisation at all. Its probably a fair bit more work though. It seems that there is a SetValidator method on Ethereum which is accessible through the api. That seems to be the sticking point for now.

I don't see it as a particularly useful api call for most people, because I can't imagine a validator node changing its address during execution, since validators need to be elected and different validator addresses will be owned by different people so validator addresses won't change during the lifetime of a node.

I think in fact it is likely to catch people out because setting it via a call to setEtherbase witll set the validator on the eth service and calling the corresponding ehterbase api method will then return that value, but the new validator address won't have any effect on the backend until it the backend is stopped and started  by calls to stop and start mining. There doesn't seem to be any documentation covering that or the fact that for the restart to work you need to have imported all the new keys into the node's keystore otherwise calls to start mining will fail.

But if we could remove the concept of setting the validator then we could set this once at construction time and not have to worry about synchronisation at all.

Reply:
I 100% agree on this, and I also think that it would lead to a better codebase. However I want to keep the focus on the actual motive for this PR, which is making it easier to write a clean AnnounceManager. And with our without your proposed changes (which I agree on), we'll still need probably a way to group these fields together instead of having them hanging in the istanbul Backend. The atomic would be moot, that's for sure, though it's definitely not the focus of this PR, but the ill named `AuthorizeInfo` is.

Reply:
Sure, I'm definitely for the grouping. I just wanted to raise the idea, I wasn't expecting to roll it into this PR, I should have been more clear about that. But it's good to know that you concur :)